src/imitation/data/rollout.py

```diff
@@ -358,10 +358,16 @@ def rollout_stats(trajectories: Sequence[types.TrajectoryWithRew]) -> Dict[str,
         "len": np.asarray([len(t.rews) for t in trajectories]),
     }
 
-    infos_peek = trajectories[0].infos
-    if infos_peek is not None and "episode" in infos_peek[-1]:
-        monitor_ep_returns = [t.infos[-1]["episode"]["r"] for t in trajectories]
+    monitor_ep_returns = []
+    for t in trajectories:
+        if t.infos is not None:
+            ep_return = t.infos[-1].get("episode", {}).get("r")
+            if ep_return:
+                monitor_ep_returns.append(ep_return)
+    if monitor_ep_returns:
         traj_descriptors["monitor_return"] = np.asarray(monitor_ep_returns)
+        # monitor_return_len may be < n_traj when infos is sometimes missing
+        out_stats["monitor_return_len"] = len(traj_descriptors["monitor_return"])
```

Comment:
It feels potentially confusing that the i-th `monitor_return` doesn't necessarily correspond to the same episode as the i-th `return` or `len`. I'm not sure if there's a better way though; we could use masked numpy arrays but I'm not familiar enough with them to say whether that would just lead to more issues down the road. So probably just leave it as it is but add a sentence about it to the docstring?